### PR TITLE
Implemented HeadingFinder in ConvoyLeader Application

### PR DESCRIPTION
**Pull request by gabryelreyes**

### CODE REVIEW

File: lib/APPConvoyLeader/App.cpp [hunk 1]
```diff
@@ -112,14 +127,35 @@ void App::loop()
...
+    OdometryData payload;
+    int32_t      xPos = 0;
+    int32_t      yPos = 0;
+
+    odometry.getPosition(xPos, yPos);
+    payload.xPos        = xPos;
+    payload.yPos        = yPos;
+    payload.orientation = odometry.getOrientation();
+
+    m_smpServer.sendData(m_serialMuxProtChannelIdOdometry, reinterpret_cast<uint8_t*>(&payload), sizeof(payload));
```
Comment (BlueAndi) [inline lib/APPConvoyLeader/App.cpp:143]: If return value is not handled, please make this explict visible.

File: lib/APPConvoyLeader/App.cpp [hunk 2]
```diff
@@ -112,14 +127,35 @@ void App::loop()
...
+
+    display.clear();
+    display.print("Heading:");
+    display.gotoXY(0U, 1U);
+    display.print(payload.orientation);
```
Comment (BlueAndi) [inline lib/APPConvoyLeader/App.cpp:148]: Consider that writing to the display might be slow on target.

File: lib/APPConvoyLeader/App.cpp [hunk 3]
```diff
@@ -112,14 +127,35 @@ void App::loop()
...
-void App::positionCallback(const uint8_t* payload, const uint8_t payloadSize)
+void App::reportSpeed()
 {
-    ; /* Do nothing. */
+    Speedometer& speedometer = Speedometer::getInstance();
+    SpeedData    payload;
+    payload.left  = speedometer.getLinearSpeedLeft();
+    payload.right = speedometer.getLinearSpeedRight();
+
+    m_smpServer.sendData(m_serialMuxProtChannelIdSpeed, reinterpret_cast<uint8_t*>(&payload), sizeof(payload));
```
Comment (BlueAndi) [inline lib/APPConvoyLeader/App.cpp:158]: If return value is not handled, please make this explict visible.

File: lib/APPConvoyLeader/MotorSpeedCalibrationState.cpp [hunk 4]
```diff
@@ -218,7 +218,7 @@ void MotorSpeedCalibrationState::finishCalibration(StateMachine& sm)
         LOG_DEBUG_VAL(TAG, "Calibrated max. speed (steps/s): ", maxSpeed);
         LOG_DEBUG_VAL(TAG, "Calibrated max. speed (mm/s): ", maxSpeed / RobotConstants::ENCODER_STEPS_PER_MM);
 
-        sm.setState(&LineSensorsCalibrationState::getInstance());
+        sm.setState(&ReadyState::getInstance());
```
Comment (BlueAndi) [inline lib/APPConvoyLeader/MotorSpeedCalibrationState.cpp:221]: Does this mean the line sensors are not used at all or just for now?

**Review by BlueAndi — CHANGES_REQUESTED** [review]

File: lib/APPConvoyLeader/MotorSpeedCalibrationState.cpp (hunk 4 above)
Comment (gabryelreyes) [reply lib/APPConvoyLeader/MotorSpeedCalibrationState.cpp:221]: Just for now. They are not relevant to the application yet, but will be used in the future.
Removing the state altogether would be a waste of time if it has to be brought back at a later point.

File: lib/APPConvoyLeader/SerialMuxChannels.h [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+
+/** Struct of the "Odometry" channel payload. */
+typedef struct _OdometryData
+{
+    int32_t xPos;        /**< X position. */
+    int32_t yPos;        /**< Y position. */
+    int32_t orientation; /**< Orientation. */
```
Comment (BlueAndi) [inline lib/APPConvoyLeader/SerialMuxChannels.h:82]: Units!

File: lib/APPConvoyLeader/SerialMuxChannels.h [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+} __attribute__((packed)) OdometryData;
+
+/** Struct of the "Speed" channel payload. */
+typedef struct _SpeedData
+{
+    int16_t left;  /**< Left motor speed. */
```
Comment (BlueAndi) [inline lib/APPConvoyLeader/SerialMuxChannels.h:88]: Units!

File: lib/APPConvoyLeader/SerialMuxChannels.h [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    int16_t right; /**< Right motor speed. */
```
Comment (BlueAndi) [inline lib/APPConvoyLeader/SerialMuxChannels.h:89]: Units!

**Review by BlueAndi — CHANGES_REQUESTED** [review]

**Review by BlueAndi — APPROVED** [review]

**Pull request merged**